### PR TITLE
Refactor SparseSliceGradOp to use a functor

### PR DESCRIPTION
**Pull request by benbarsdell**
- This is in preparation for adding a GPU implementation.
- No functional change.
- Also adds some explicit tests.

cc @nluehr

### CODE REVIEW

**Comment by gbaned** [comment]
@penpornk Can you please review this PR ? Thanks!

**Comment by gbaned** [comment]
@benbarsdell Can you please resolve conflicts? Thanks!

**Comment by benbarsdell** [comment]
Rebased (strangely I didn't get any conflicts).

**Comment by benbarsdell** [comment]
Rebased again.

File: tensorflow/python/kernel_tests/sparse_ops/sparse_slice_op_test.py [hunk 1]
```diff
@@ -283,6 +283,44 @@ def testGradients(self):
...
+    with test_util.force_cpu():
+      start, size = [0, 0], [4, 1]
+      sp_output = sparse_ops.sparse_slice(sp_input, start, size)
+      input_grad_vals = sparse_ops.sparse_slice_grad(
+          sp_output.values, sp_input.indices, start, sp_output.indices)
+      self.assertAllEqual(input_grad_vals, [0, 0, 0, 0,
+                                            0, 0, 0,
+                                            20, 0, 0,
+                                            30, 0, 0, 0])
+
```
Comment (penpornk) [inline tensorflow/python/kernel_tests/sparse_ops/sparse_slice_op_test.py:298]: Thank you for formatting them this way! This is easy to read.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and I'm so, so sorry for the delay!
I have one nit.

**Review by penpornk — APPROVED** [review]
Thank you again for your patience!

**Comment by benbarsdell** [comment]
Rebased.

**Review by penpornk — APPROVED** [review]
Thank you for rebasing!

**Comment by benbarsdell** [comment]
Out of interest, what's up with all the `FUTURE_COPYBARA_INTEGRATE_REVIEW=https://github.com/tensorflow/tensorflow/pull/52007` in unrelated commits?

**Pull request merged**